Support PR: Adding new version-awareness to the CLI export:workflow command (#4541)
n8n-io/n8n-docs: docs/hosting/cli-commands.md

n8n execute --id <ID>
```

## Publish or unpublish a workflow

You can publish or unpublish a workflow using the CLI. In n8n 2.0, the [previous active/inactive toggle](/2-0-breaking-changes.md) was replaced by a publish/unpublish model. Use `publish:workflow` and `unpublish:workflow` to change a workflow's published state from the CLI.

/// note | Restart required
These commands operate on your n8n database. If you execute them while n8n is running, the changes don't take effect until you restart n8n.
///

### Publish a workflow

Use `publish:workflow` to publish a workflow by its ID. You can optionally publish a specific historical version by passing its `versionId`.

Command flags:

| Flag | Description |
|------|-------------|
| --help | Help prompt. |
| --id | The ID of the workflow to publish. Required. |
| --versionId | Optional version ID to publish. If omitted, the current draft is published. |

/// note | No `--all` flag

> Unlike the deprecated `update:workflow` command, `publish:workflow` does not support `--all`. This is intentional: it prevents accidental bulk publishing of workflows in production environments. Publish workflows individually by ID.

Reviewer: ```suggestion
Unlike the deprecated `update:workflow` command, `publish:workflow` doesn't support `--all`. This is intentional: it prevents accidental bulk publishing of workflows in production environments. Publish workflows individually by ID.
```

```bash
n8n export:workflow --backup --output=backups/latest/
```
#### Export a specific workflow version

You can export a specific historical version of a workflow by passing its `versionId` with `--version`:

```bash
n8n export:workflow --id=<ID> --version=<VERSION_ID> --output=workflow-v1.json
```

#### Export the published version of a workflow

Use `--published` to export the currently published/active version of a workflow rather than the current draft:

```bash
n8n export:workflow --id=<ID> --published --output=published.json
```

You can combine `--published` with `--all` to export every workflow's published version. Workflows that don't have a published version are skipped:

```bash
n8n export:workflow --all --published --output=workflows.json
```

/// note | Version metadata

> When exporting a workflow, n8n includes a `versionMetadata` property containing the workflow's historical name and description for that version. The import command preserves this data in the workflow history table on import. The current workflow's name and description are not overridden.

Reviewer: ```suggestion
When exporting a workflow, n8n includes a `versionMetadata` property containing the workflow's historical name and description for that version. The import command preserves this data in the workflow history table on import. The current workflow's name and description aren't overridden.
```
